screens/account/VerifyEmail.tsx

```diff
@@ -1,28 +1,26 @@
-import { Button, Center, Heading, Text, VStack } from 'native-base';
-import { useEffect } from 'react';
-import { useRecoilState, useSetRecoilState } from 'recoil';
-import {
-  isEmailVerifiedAtom,
-  userPermissionLevelAtom,
-} from '../../utils/atoms';
-import { sendAuthEmail, startWaitForVerificationPoll } from '../../xplat/api';
+import { Button, Center, Heading, Input, Text, VStack } from 'native-base';
+import { useEffect, useState } from 'react';
+import { useSetRecoilState } from 'recoil';
+import { userPermissionLevelAtom } from '../../utils/atoms';
 import { auth } from '../../xplat/Firebase';
-import { UserStatus } from '../../xplat/types';
+import { confirmEmailCode, sendEmailCode } from '../../xplat/api';
 
 const VerifyEmail = () => {
-  const setIsEmailVerified = useSetRecoilState(isEmailVerifiedAtom);
-  const [userPermissionLevel, setUserPermissionLevel] = useRecoilState(
-    userPermissionLevelAtom
-  );
+  const setUserPermissionLevel = useSetRecoilState(userPermissionLevelAtom);
+  const [targCode, setTargCode] = useState<number>();
+  const [attemptCode, setAttemptCode] = useState<string>('');
+
+  useEffect(() => {
+    if (!targCode) sendEmailCode().then(setTargCode);
+  }, [targCode]);
 
   useEffect(() => {
```

Comment:
@tylerhm I'd like to draw your attention specifically to this useEffect since you know the most about this virtual component bullshit. Is this gonna call `sendEmailCode` sometimes when we don't actually physically render the `VerifyEmail` component? If so, is there a way to fix this?

Reply:
It shouldn't, but there is always some fuckery, so to be safe we should add a check for the user's permission level in the `useState`

```ts
if (userPermissionLevel !== undefined && userPermissionLevel <= UserStatus.Unverified) {
  ...
}
```